libbpfgo.go

```diff
@@ -365,9 +366,10 @@ func (m *Module) GetMap(mapName string) (*BPFMap, error) {
 	}
 
 	return &BPFMap{
-		name:   mapName,
-		fd:     C.bpf_map__fd(bpfMap),
-		module: m,
+		definition: C.bpf_map__def(bpfMap),
```

Comment:
What do you think about exposing API for getting the map definition? We have one for getting max entries right now which is only part of the map definition.

Reply:
That would be strictly additional to what's in this PR no? Happy to discuss what this should look like in an issue and do a follow-up.

Reply:
Correct, it could be a follow up PR, that likely wouldn't change anything in the code you changed here.

Reply:
Actually, I take back my suggestion. I'm re-reading the [libbpf 1.0 roadmap](https://docs.google.com/document/d/1UyjTZuPFWiPFyKk1tV5an11_iaRuec6U-ZESZ54nNTY/edit#), and it states that using `bpf_map__def()` should be discouraged in favor of getter APIs for the individual fields of the bpf map definition. In this case we can just use [`bpf_map__value_size()`](https://github.com/libbpf/libbpf/blob/master/src/libbpf.h#L459) here.

I'm making issues for the individual map definition APIs now!